modules/help_editor/jsx/help_editor.js

```diff
@@ -0,0 +1,168 @@
+ *
+ * Serves as an entry-point to the module, rendering the whole react
+ * component page on load.
+ *
+ * Renders Help Editor main page consisting of FilterTable and
+ * DataTable components.
+ *
+ * @author LORIS Team
+ * @version 1.0.0
+ *
+ * */
+class HelpEditor extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isLoaded: false,
+      filter: {},
+      data: {}
+    };
+
+    /*
+     * Bind component instance to custom methods
+     */
+    this.fetchData    = this.fetchData.bind(this);
+    this.updateFilter = this.updateFilter.bind(this);
+    this.resetFilters = this.resetFilters.bind(this);
+    this.formatColumn = this.formatColumn.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  /**
+   * Retrieve data from the provided URL and save it in state
+   */
+  fetchData() {
+    $.ajax(this.props.DataURL, {
+      method: 'GET',
+      dataType: 'json',
+      success: data => {
+	      this.setState({
+          data: data,
+          isLoaded: true
+        });
+      },
+      error: error => console.error(error)
+    });
+  }
+
+  /**
+   * Set this.state.filter to the input filter object
+   *
+   * @param {object} filter - the filter object
+   */
+  updateFilter(filter) {
+    this.setState({filter});
+  }
+
+  //TODO: Clearing filters via refs should be deprecated in future refactoring.
+  /**
+   * Reset the filter elements with textInput refs to empty values
+   */
+  resetFilters() {
+    this.filter.clearFilter();
+  }
+
+  /**
+   * Modify behaviour of specified column cells in the Data Table component
+   *
+   * @param {string} column - column name
+   * @param {string} cell - cell content
+   * @param {arrray} rowData - array of cell contents for a specific row
+   * @param {arrray} rowHeaders - array of table headers (column names)
+   *
+   * @return {*} a formated table cell for a given column
+   */
+  formatColumn(column, cell, rowData, rowHeaders) {
+    if (this.state.data.hiddenHeaders.indexOf(column) > -1) {
+      return null;
+    }
+
+    // Create the mapping between rowHeaders and rowData in a row object.
+    let row = {};
+    let url;
+    rowHeaders.forEach((header, index) => row[header] = rowData[index]);
+  
+    if (column === 'Topic') {
+      url = loris.BaseURL + '/help_editor/edit_help_content/?helpID=' +
+             row.HelpID + '&parentID=' + row.ParentID;
+      return <td><a href ={url}>{cell}</a></td>;
+    }
+    if (column === 'Parent Topic') {
+      url = loris.BaseURL + '/help_editor/edit_help_content/?helpID=' +
+             row.ParentID + '&parentID=' + row.ParentTopicID;
+      return <td><a href ={url}>{cell}</a></td>;
+    }
+  
+    return <td>{cell}</td>;
+  }
+
+  render() {
+    // Waiting for async data to load
+    if (!this.state.isLoaded) {
+      return <Loader/>
+    }
+
+    //TODO: refs should be deprecated in future refactoring.
+    const filterRef = function(f) {
```

Comment:
this could maybe be put in the constructor as it doesn't need to be re-rendered?

Reply:
it wouldn't have to be set as a state in order to be accessed within the render function. from this document (https://reactjs.org/docs/refs-and-the-dom.html#callback-refs), the ref call back method that you are using here is set within the constructor. i have also implemented in my PR (https://github.com/aces/Loris/pull/3924) on line (https://github.com/aces/Loris/blob/46bce3d252cfd3a153741f5a9e196794ee22fbdd/modules/examiner/jsx/examiner.js#L38). such a function doesn't belonging in the rendering of the DOM

Reply:
Hmm, well based on React's own guidelines for how refs should be used (https://reactjs.org/docs/refs-and-the-dom.html#when-to-use-refs), we shouldn't be using callback refs in this case at all. I didn't introduce this line of code and I rather not move it around, so I think it's best to simply refactor the code now.